app/Http/Controllers/EventController.php

```diff
@@ -16,8 +16,11 @@ class EventController extends Controller
      */
     public function index(Request $request)
     {
-        $events = Event::all();
+        $events = Event::whereRegistered()
```

Comment:
if you have this here, it means that we're *always* filtering the events by if a user is registered or not. What you'd want to do is check for a flag like `$request->has('registered_only')` or what ever name you want in an if statement, if it has that then you add the `whereRegistered()` call, otherwise don't add it